titus-api/src/main/java/com/netflix/titus/api/eviction/model/event/EvictionEvent.java

```diff
@@ -0,0 +1,40 @@
+package com.netflix.titus.api.eviction.model.event;
+
+import com.netflix.titus.api.eviction.model.EvictionQuota;
+import com.netflix.titus.api.eviction.model.SystemDisruptionBudget;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.api.model.ApplicationSLA;
+import com.netflix.titus.api.model.Reference;
+import com.netflix.titus.api.model.Tier;
+import com.netflix.titus.common.util.limiter.tokenbucket.TokenBucket;
+
+public abstract class EvictionEvent {
```

Comment:
`interface`?

Reply:
I would prefer that this class be an interface and you have a separate `EvictionEvents` class for factory methods unless I am missing something.

Reply:
We use abstract base classes for all event hierarchies.